src/polymer/declaration-property-handlers.ts

```diff
@@ -16,15 +16,48 @@ import * as estree from 'estree';
 
 import * as astValue from '../javascript/ast-value';
 import {JavaScriptDocument} from '../javascript/javascript-document';
-import {Severity} from '../warning/warning';
+import {ScannedBehaviorAssignment} from '../polymer/behavior';
+import {Severity, Warning} from '../warning/warning';
 
 import {analyzeProperties} from './analyze-properties';
 import {ScannedPolymerElement} from './polymer-element';
 
+export type BehaviorAssignmentOrWarning = {
+  kind: 'warning',
+  warning: Warning
+} | {kind: 'behaviorAssignment', assignment: ScannedBehaviorAssignment};
+
+export function getBehaviorAssignmentOrWarning(
```

Comment:
This design with two possible return types is interesting to see in TypeScript. I think it's fine as written, but in the future it might be more clear to only return a single behavior assignment type on success, and throw an exception on failure with the `warning` data on it.